src/librustc_resolve/lib.rs

```diff
@@ -879,7 +879,7 @@ pub struct Resolver<'a> {
     /// `CrateNum` resolutions of `extern crate` items.
     extern_crate_map: FxHashMap<LocalDefId, CrateNum>,
     export_map: ExportMap<LocalDefId>,
-    trait_map: TraitMap<NodeId>,
+    trait_map: NodeMap<Vec<TraitCandidate>>,
```

Comment:
If you use a `NodeMap` here, it means the keys are still `NodeId` and that you still need to call `node_id_to_hir_id` later.